src/query/frontend/ast/cypher_main_visitor.cpp

```diff
@@ -492,47 +496,281 @@ antlrcpp::Any CypherMainVisitor::visitStreamQuery(MemgraphCypher::StreamQueryCon
 }
 
 antlrcpp::Any CypherMainVisitor::visitCreateStream(MemgraphCypher::CreateStreamContext *ctx) {
+  MG_ASSERT(ctx->children.size() == 1, "CreateStreamQuery should have exactly one child!");
+  auto *stream_query = ctx->children[0]->accept(this).as<StreamQuery *>();
+  query_ = stream_query;
+  return stream_query;
+}
+
+namespace {
+std::vector<std::string> TopicNamesFromSymbols(
+    antlr4::tree::ParseTreeVisitor &visitor,
+    const std::vector<MemgraphCypher::SymbolicNameWithDotsAndMinusContext *> &topic_name_symbols) {
+  MG_ASSERT(!topic_name_symbols.empty());
+  std::vector<std::string> topic_names;
+  topic_names.reserve(topic_names.size());
+  std::transform(topic_name_symbols.begin(), topic_name_symbols.end(), std::back_inserter(topic_names),
+                 [&visitor](auto *topic_name) { return JoinSymbolicNamesWithDotsAndMinus(visitor, *topic_name); });
+  return topic_names;
+}
+
+template <bool required, typename... ValueTypes>
+void MapConfig(auto &memory, const auto &enum_key, auto &destination) {
+  const auto key = static_cast<uint8_t>(enum_key);
```

Comment:
I'm not sure why didn't you use `std::underlying_type_t` for enum_key?

Reply:
```
  template <typename T>
  concept EnumUint8 = std::is_enum_v<T> && std::same_as<uint8_t, std::underlying_type_t<T>>;

  template <bool required, typename... ValueTypes>
  void MapConfig(auto &memory, const EnumUint8 auto enum_key, auto &destination) {
```
I would do something like that. I used concept with auto so we don't have to specify enum type as a template argument. It cannot be deduced because of variadic templates.

Reply:
Sorry, I wanted to write `std::underlying_type_t<decltype(enum_key)>` instead of just `decltype(enum_key)`.